DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs

```diff
@@ -153,6 +155,58 @@ public IActionResult Diagnostic(int customisationId, int sectionId)
             return View("Diagnostic/Diagnostic", model);
         }
 
+        [Route("/LearningMenu/{customisationId:int}/{sectionId:int}/Diagnostic/Content")]
+        public IActionResult DiagnosticContent(int customisationId, int sectionId, List<int> checkedTutorials)
+        {
+            var candidateId = User.GetCandidateId();
+            var centreId = User.GetCentreId();
+            var diagnosticContent = diagnosticAssessmentService.GetDiagnosticContent(customisationId, sectionId);
+
+            if (diagnosticContent == null || centreId == null)
+            {
+                logger.LogError(
+                    "Redirecting to 404 as section/centre id was not found. " +
+                    $"Candidate id: {candidateId}, customisation id: {customisationId}, " +
+                    $"centre id: {centreId?.ToString() ?? "null"}, section id: {sectionId}");
+                return RedirectToAction("StatusCode", "LearningSolutions", new { code = 404 });
+            }
+
+            if (checkedTutorials.Except(diagnosticContent.Tutorials).Any())
+            {
+                logger.LogError(
+                    "Redirecting to 404 as tutorial id was not found. " +

```

Comment:
This error message doesn't really match what's happened. The issue is that the checked tutorials don't match the tutorials in the diagnostic assessment 🙂